### PR TITLE
ci: Add dependabot

### PR DESCRIPTION
**Pull request by andrzejnovak**
@eduardo-rodrigues

### CODE REVIEW

**Comment by andrzejnovak** [comment]
Not sure what to do here about the missing token

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# dependabot.yml
+
+version: 2
+updates:
+
+  # Maintain dependencies for GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (eduardo-rodrigues) [inline .github/dependabot.yml:10]:
```suggestion
      interval: "monthly"
```

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    reviewers:
+      - "eduardo-rodrigues"
+      - "andrzejnovak"
+    prefix: "chore(deps): "
+
+  # Maintain dependencies for python
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (eduardo-rodrigues) [inline .github/dependabot.yml:20]:
```suggestion
      interval: "monthly"
```

File: .github/dependabot.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  # Maintain dependencies for GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    reviewers:
+      - "eduardo-rodrigues"
+      - "andrzejnovak"
+    prefix: "chore(deps): "
```
Comment (eduardo-rodrigues) [inline .github/dependabot.yml:14]:
```suggestion
    prefix: "chore(deps): "
    ignore:
      - dependency-name: "actions/*"
        update-types: ["version-update:semver-minor", "version-update:semver-patch"]
```

Comment (eduardo-rodrigues) [reply .github/dependabot.yml:14]: This is what we've set up in several packages, see for example https://github.com/scikit-hep/particle/blob/master/.github/dependabot.yml.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  # Maintain dependencies for python
+  - package-ecosystem: "pip"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    reviewers:
+      - "eduardo-rodrigues"
+      - "andrzejnovak"
+    prefix: "chore(deps): "
```
Comment (eduardo-rodrigues) [inline .github/dependabot.yml:27]:
```suggestion
    prefix: "chore(deps): "
    ignore:
      - dependency-name: "*"
        update-types: ["version-update:semver-patch"]
```

**Comment by eduardo-rodrigues** [comment]
OK, seems we are progressing. But now I'm seeing the message

> Running on Github Actions but GITHUB_TOKEN is not set. Add "env: GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}" to your step config.
...

@henryiii likely knows how to work this out. I wonder where to get this since other dependabot.yml files in other org packages do not need it :S.

**Comment by henryiii** [comment]
The problem isn't dependabot, it's coveralls, which seems to have changed or broken over the last 10 months. It looks like we are passing it a token, but not the one it (now) wants.

**Comment by henryiii** [comment]
This will likely not do anything on the Python side, since we are misusing requirements.txt. It's designed to update a lock file with pins or pins with hashes, not version ranges. I think we'd need to write a little custom action to update package versions in the form we have them in. But the actions updates are good (and we can try the Python ones to see what it does!)

**Comment by eduardo-rodrigues** [comment]
Fair enough @henryiii. A matter of seeing what it does, as you say.

I see you got PR https://github.com/scikit-hep/scikit-hep/pull/179 in, in the meantime. now the CI error with Py 3.8 is different. Can you rebase @andrzejnovak?

**Comment by andrzejnovak** [comment]
@eduardo-rodrigues sure, can you enable the github button to do it? :) [link](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-rebasing-for-pull-requests)

**Comment by henryiii** [comment]
I added the button.

**Comment by eduardo-rodrigues** [comment]
Hmm, same issue. £I don't see why it should file given that the previous MR from Henry went just fine. I'm tempted to merge and see if the problem remains! Thoughts?

**Comment by andrzejnovak** [comment]
I would be in favour. The failure is not really related to the PR

**Comment by henryiii** [comment]
I think the problem is it's trying to push from a fork. Not sure how it's supposed to work for PRs, but I think that's why it's having issues.

**Comment by eduardo-rodrigues** [comment]
I fixed the file in https://github.com/scikit-hep/scikit-hep/commit/96d8c170cea72d5839cb5da564d1dff0b4c56742. Turns out the `prefix` property was only valid as a detail to `commit-message`.

**Comment by eduardo-rodrigues** [comment]
OK, dependabot is now creating a whole bunch of PRs for patch version updates, whereas only minor and major updates should be picked up. There's still some fine tuning to be done.

**Comment by henryiii** [comment]
I think we can have it ignore patch updates. Then probably should have it check more often so that it makes a PR as soon as a minor version is released.

**Pull request merged**